source/Patches/EndGameSummary.cs

```diff
@@ -2,23 +2,29 @@
 using System.Collections.Generic;
 using System.Text;
 using UnityEngine;
+using TownOfUs.Extensions;
 using TownOfUs.Roles;
 using TownOfUs.Roles.Modifiers;
 using HarmonyLib;
+using System.Linq;
 
 namespace TownOfUs.Patches
 {
     [HarmonyPatch]
     static class EndGameSummary
     {
-        static class AdditionalTempData
+        public static class AdditionalTempData
         {
+            public static bool updateFlag = false;
             public static List<PlayerInfo> playerData = new List<PlayerInfo>();
+            public static List<KeyValuePair<byte, Role>> roleHistory = new List<KeyValuePair<byte, Role>>();
+            public static List<Tuple<string, string>> summaryTextLines = new List<Tuple<string, string>>();
```

Comment:
Do we need to hang onto these? Can we not just generate them when we need them?

Reply:
Also what are the .item1 and .item2 here?

Reply:
> Also what are the .item1 and .item2 here?

Tuple is a very basic form of "key" and "value" pair.

Item1 is Key
Item2 is pair

its been so long since i wrote this, but for whatever reason i had to resort to a tuple as the keyvaluepair was not working for me.

Reply:
> Do we need to hang onto these? Can we not just generate them when we need them?

this generates them when the game is loaded?
it clears (empties) them when the game starts.
it fills them over the duration of the game.

i think i had issues with generating them at the start of each game, not really sure tbh

Reply:
Yes, I know what a Tuple is. My question is what the value1 and value2 actually represent.

Reply:
It's been so long since I updated this I had to re look at the code.

Item1 is a value determining if a player won the game.

Item2 is the string representing the summary table row.

I added the lines to this list so they could be sorted by Items, so all winners are placed at the top.

There are other ways of doing this, but I figured this was the easiest way. Creating 2 seperate lists is another way but considering there may be 13 winners or 7 winners or 2 winners, or 1 winner, or Any amount of winners, I figured it was easier to sort the current list of lines.

As winning/losing is not added until just before it is displayed, I still think this was the simplest way.

Reply:
> It's been so long since I updated this I had to re look at the code
That’s kind of the issues with Tuple in a nutshell. Especially when you put the same type in both T1 and T2. It’s a very very convenient tool, but it becomes devoid of context _very_ fast.

Reply:
> > It's been so long since I updated this I had to re look at the code
> > That’s kind of the issues with Tuple in a nutshell. Especially when you put the same type in both T1 and T2. It’s a very very convenient tool, but it becomes devoid of context _very_ fast.

No I get that, but all it takes is a comment to explain, I used tuple because I had to afaik. If I can change it I will.

Reply:
Replaced Tuple with KeyValuePair.. seems to be still working.